contractcourt/interfaces.go

```diff
@@ -65,3 +66,65 @@ type UtxoSweeper interface {
 	UpdateParams(input wire.OutPoint, params sweep.ParamsUpdate) (
 		chan sweep.Result, error)
 }
+
+// InterceptedPacket contains the relevant information for the interceptor about
+// an htlc.
+type InterceptedPacket struct {
```

Comment:
this move feels semantically off, why is it needed?

Reply:
Yes, dependency cycles indeed. `contractcourt` cannot access `htlcswitch` because of that. Good idea to make a package for it.

Reply:
Will fix it in a follow up PR. Changes and moves in the same PR don't go together very well. Unless you'd be happy with a skeleton package that contains just the types and move over the rest of the interceptor code in the follow up? Also think about the package path. `/interceptor`? We don't have a common parent directory for the on and offchain flows.

Reply:
Created